cherrypy/_cpreqbody.py

```diff
@@ -724,14 +738,22 @@ def default_proc(self):
                 self.file = result
 
     def read_into_file(self, fp_out=None):
-        """Read the request body into fp_out (or make_file() if None).
+        """Read the request body into fp_out.
+
+        if fp_out not given, use either make_file or
+        a BytesIO object, based on maxrambytes.
 
         Return fp_out.
         """
-        if fp_out is None:
-            fp_out = self.make_file()
-        self.read_lines_to_boundary(fp_out=fp_out)
-        return fp_out
+        if not fp_out:
+            x = self.read_lines_to_boundary()
+            if isinstance(x,(str,bytes)):
```

Comment:
So just check if the BytesIO constructor fails due to getting a file-like object? I like the explicitness of checking for the specific type, but it does kind of make it look like x might be an actual unicode string if you don't know the python2 history.

Either way there should probably be a comment there I'd say.